iroh/src/client/blobs.rs

```diff
@@ -360,18 +367,37 @@ where
 
     /// Read the content of a collection.
     pub async fn get_collection(&self, hash: Hash) -> Result<Collection> {
-        let BlobGetCollectionResponse { collection } =
-            self.rpc.rpc(BlobGetCollectionRequest { hash }).await??;
-        Ok(collection)
+        Ok(Collection::load(hash, self).await?)
     }
 
     /// List all collections.
     pub async fn list_collections(&self) -> Result<impl Stream<Item = Result<CollectionInfo>>> {
-        let stream = self
-            .rpc
-            .server_streaming(BlobListCollectionsRequest)
-            .await?;
-        Ok(flatten(stream))
+        let this = self.clone();
+        Ok(Gen::new(|co| async move {
+            if let Err(cause) = this.list_collections_impl(&co).await {
+                co.yield_(Err(cause)).await;
+            }
+        }))
+    }
+
+    async fn list_collections_impl(&self, co: &Co<Result<CollectionInfo>>) -> Result<()> {
+        let tags = self.tags_client();
+        let mut tags = tags.list().await?;
```

Comment:
I wonder if we want to add a `tags.list(BlobFormat)`, given with a large db this would be a lot of queries

Reply:
We could, but you should not have that many tags in the first place. All the "short lived" tag use should move to temp tags.

In any case I moved this stuff in a separate PR